### ISSUE TITLE
Prüfung der erlaubten Dateiformate im Admin-Filemanager

### ISSUE THREAD

**Issue by alw-bsz** [OPUS4/application#524]
Die zulässigen Dateiformate (Parameter publish.filetypes.allowed) sollten auch beim Upload über den Admin-Filemanager geprüft und nicht-zulässige Formate abgewiesen werden. Damit Instanzen, in denen das gewohnte Verhalten (Upload aller Dateiformate zulässig) erhalten bleiben soll, weiterhin so gefahren werden können, ist es vermutlich sinnvoll, die Prüfung optional zu machen. Es wären daher wohl zwei Erweiterungen nötig:

- Ergänzen der application.ini um einen Parameter, um die Prüfung der Dateiformate im Admin-Filemanager an- bzw. auszuschalten
- Erweitern der Dateiprüfung im Admin-Filemanager um die Dateiformate

**Comment by j3nsch** [comment]
Es geht hier um die Prüfung im Browser, also vor dem Upload richtig?

Intern: https://tickets.zib.de/jira/browse/OPUSVIER-4118

**Comment by alw-bsz** [comment]
Nein, es geht darum, dass beim Upload über admin/filemanager/upload nicht geprüft wird, ob es sich um ein zulässiges Dateiformat handelt und man Dateien jeden beliebigen Typs hochladen kann.

Beim Upload im Publish-Modul wird auf die Parameter Dateiformat sowie Zeichen und Länge des Dateinamens geprüft, im Admin-Modul nur auf Zeichen und Länge des Dateinamens.

Vgl. vermutlich /application/modules/admin/forms/File/Upload.php (Z. 83ff.) :
```
        $filenameOptions = [
            'filenameMaxLength' => $config->publish->filenameMaxLength,
            'filenameFormat' => $config->publish->filenameFormat
        ];
        $filenameValidator = new Application_Form_Validate_Filename($filenameOptions);
```

**Comment by j3nsch** [comment]
Ja, die Prüfungen im Publish-Modul wurden vor ein paar Jahren hinzugefügt. Für die Administration galt die Regel, dass Admins wissen sollten was sie tun und deshalb existieren dort kaum Einschränkungen.

Die Prüfung kann hinzugefügt werden, hat aber aus meiner Sicht keine hohe Priorität. Das können wir machen, wenn wir den Filemanager auch aus anderen Gründen anfassen müssen.
